cli/cmd/common/utils.go

```diff
@@ -1,20 +1,81 @@
 package common
 
 import (
+	"embed"
 	"fmt"
 	"os"
 	"strings"
 
 	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/chart/loader"
 	helmCLI "helm.sh/helm/v3/pkg/cli"
 	"k8s.io/cli-runtime/pkg/genericclioptions"
 )
 
 const (
 	DefaultReleaseName      = "consul"
 	DefaultReleaseNamespace = "consul"
+	chartFileName           = "Chart.yaml"
+	valuesFileName          = "values.yaml"
+	templatesDirName        = "templates"
+	TopLevelChartDirName    = "consul"
 )
 
+// ReadChartFiles reads the chart files from the embedded FS, and loads their contents into []*loader.BufferedFile. This
+// is a format that the Helm Go SDK functions can read from to create a chart to install from. The names of these files
+// are important, as there are case statements in the Helm Go SDK looking for files named "Chart.yaml" or
+// "templates/<templatename>.yaml", which is why even though the embedded FS has them named "consul/Chart.yaml" we have
+// to strip the "consul" prefix out.
+func ReadChartFiles(chart embed.FS, chartDirName string) ([]*loader.BufferedFile, error) {
+	var chartFiles []*loader.BufferedFile
+
+	bytes, err := chart.ReadFile(fmt.Sprintf("%s/%s", chartDirName, chartFileName))
+	if err != nil {
+		return []*loader.BufferedFile{}, err
```

Comment:
Here and below can you just return the empty `chartFiles` you instantiated at the top of the file?

Reply:
In L45 if we return `chartFiles`, it would have one file in it ("Chart.yaml"), and I don't think we should return an incomplete object in an error case

Reply:
Good call

Reply:
chiming in here: imo if you're return a non-nil error then all other return values should be considered invalid and so it's okay to return whatever you want as the other values–usually what the function writer finds easier.

I can't actually find this idiom defined anywhere though :)

So for this I'd probably do `return nil, err` because it's easier to write (and also in this case it's functionally the same as `[]*loader.BufferedFile{}`.